gen3utils/etl/etl_validator.py

```diff
@@ -23,11 +23,13 @@ def __init__(self, name, underscore=False):
         self.props = {id_name: Prop(id_name)}
 
 
-def validate_joining_list_props(props_list, recorded_errors, existing_indices):
+def validate_joining_list_props(index_name, props_list, recorded_errors, existing_indices):
     if type(props_list) is list:
         for prop in props_list:
             if "index" in prop and "join_on" in prop:
                 for real_prop in prop.get("props"):
+                    if real_prop.get("name") in existing_indices.get(index_name).props:
```

Comment:
i don't understand how this works 🤔 it looks to me like for each prop in the index, you're checking if it's in the index, but it will always be in the index - how do you know it's a duplicate?

could you explain?

Reply:
the `existing_indcies.get(index_name)` point to the index that subject to check in this iterative validation. This function is to check the duplication in `joining_props`. Since we wait until the end of validation when we have all indices to validate.